docs/en/wallet-attestation.rst

```diff
@@ -30,271 +33,271 @@ The following requirements for the Wallet Attestation are met:
 
-* ``grant_type`` set to ``urn:ietf:params:oauth:grant-type:jwt-bearer``;
-* ``assertion`` containing the signed JWT defined in the Section `Wallet Attestation Request`_.
+**Steps 3-5:**: The Wallet Instance request and receive a one time ``challenge`` from the Wallet Provider Backend. The ``challenge`` is a ``nonce``, which MUST be unpredictable and it is the primary countermeasure against replay attack. The value of the ``nonce`` returned by the backend must be generated in such a way as to guarantee that it can only be used once and in a pre-established time interval.
 
-Below a non-normative example of the HTTP request.
 
 .. code-block:: http
 
-    POST /token HTTP/1.1
-    Host: wallet-provider.example.org
-    Content-Type: application/x-www-form-urlencoded
+    GET /nonce HTTP/1.1
+    Host: walletprovider.example.com
 
-    grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
-    &assertion=eyJhbGciOiJFUzI1NiIsImtpZCI6ImtoakZWTE9nRjNHeGRxd2xVTl9LWl83NTVUT1ZEbmJIaDg2TW1KcHh2a1UifQ.eyJpc3MiOiAidmJlWEprc000NXhwaHRBTm5DaUc2bUN5dVU0amZHTnpvcEd1S3ZvZ2c5YyIsICJhdWQiOiAiaHR0cHM6Ly93YWxsZXQtcHJvdmlkZXIuZXhhbXBsZS5vcmciLCAianRpIjogImY1NjUyMDcyLWFiZWYtNDU5OS1iODYzLTlhNjkwNjA3MzJjYyIsICJub25jZSI6ICIuLi4uLiIsICJjbmYiOiB7Imp3ayI6IHsiY3J2IjogIlAtMjU2IiwgImt0eSI6ICJFQyIsICJ4IjogIjRITnB0SS14cjJwanlSSktHTW56NFdtZG5RRF91SlNxNFI5NU5qOThiNDQiLCAieSI6ICJMSVpuU0IzOXZGSmhZZ1MzazdqWEU0cjMtQ29HRlF3WnRQQklScXBObHJnIiwgImtpZCI6ICJ2YmVYSmtzTTQ1eHBodEFObkNpRzZtQ3l1VTRqZkdOem9wR3VLdm9nZzljIn19LCAiaWF0IjogMTY5MTQ4ODk2MiwgImV4cCI6IDE2OTE0OTYxNjJ9.Dg_yFaiv6lVftR3FFx0v5JW250mBgXLVP1j0ezZcHRyitqSY7xGmx4y-MGur93FAS85vf_Da-L-REVEltwU2Jw
+.. code-block:: http
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json
+
+    {
+      "nonce": "d2JhY2NhbG91cmVqdWFuZGFt"
+    }
+
+**Step 6**: The Wallet Instance generates the Wallet Hardware Keys pair via the operating system and store the relative Wallet Hardware Key Tag in a local storage:
+
+   1. It MUST ensure that Wallet Hardware Keys do not already exist, if they exist and the Wallet is in the initialization phase they must be deleted.
+   2. It MUST generate a pair of asymmetric EC keys (Wallet Hardware Keys) via a local WSCD.
+   3. It SHOULD obtain a unique identifier (Wallet Hardware Key Tag) of the generated Wallet Hardware Keys from the operating system. If the operating system allows you to define a tag when creating keys, then a random string for Wallet Hardware Key Tag must be chosen.
+   4. It MUST store the relative Wallet Hardware Key Tag in a local storage.
+
+.. note::
+
+    **WSCD:** For key generation, the Wallet Instance will always rely on a local WSCD on devices that support it. For Android, Strongbox will be used or, if it is not available, a Trusted Execution Environment (TEE) while for iOS Secure Elements (SE). Since each OEM develops a different SDK for accessing the local WSCD, it will be treated in a generic manner below.
```

Comment:
```suggestion
    **WSCD:** The Wallet Instance MAY use a local WSCD for key generation on devices that support this feature. On Android devices, it will use Strongbox, or a Trusted Execution Environment (TEE) if Strongbox is unavailable. For iOS devices, Secure Elements (SE) will be employed. Given that each OEM offers a distinct SDK for accessing the local WSCD, the discussion hereafter will address this topic in a general context.
```

We cannot force the implementation in use local WSCD, we can define an impl profile using local WSCD for satisfying the requirement of LoA High withou preventing other implementation in using local external or remote  or hybrid WSCD.

Reply:
We cannot use this solution with other WSCDs that are not local to the device. The **Key Attestation** and **Integrity Check** systems are all based on the **local WSCD**. Without it it is not possible to use them.

Reply:
This technical specification is intended for both the public and the private solution and aims to provide a cohompresive guidance for the implementation where a local WSCD represent a solution and not the only solution, as previously agreed.

For this reason, it is important to do some editorial changes to make explicit the possibility to use other WSCD types, where the recommended one is the one that brings both the security and the ux at a high level.